deathnode/notebook_test.go

```diff
@@ -49,15 +87,18 @@ func TestDestroyInstanceAttempt(t *testing.T) {
 			Convey("Check remove instance protection flags", func() {
 				notebook.DestroyInstancesAttempt()
 				Convey("if it has instanceProtection flag, it should be removed", func() {
-					So(awsConn.Requests["RemoveASGInstanceProtection"], ShouldNotBeNil)
-					So(instanceMonitor.IsProtected(), ShouldBeFalse)
-				})
-				Convey("if it doesn't have instanceProtection flag, it should do nothing", func() {
-					notebook.DestroyInstancesAttempt()
 					So(awsConn.Requests["RemoveASGInstanceProtection"], ShouldNotBeNil)
 					So(len(awsConn.Requests["RemoveASGInstanceProtection"]), ShouldEqual, 1)
 					So(instanceMonitor.IsProtected(), ShouldBeFalse)
 				})
+				Convey("if it doesn't have instanceProtection flag", func() {
+					Convey("it should not remove more instanceProtection flag", func() {
+						notebook.DestroyInstancesAttempt()
+						So(awsConn.Requests["RemoveASGInstanceProtection"], ShouldNotBeNil)
+						So(len(awsConn.Requests["RemoveASGInstanceProtection"]), ShouldEqual, 1)
+						So(instanceMonitor.IsProtected(), ShouldBeFalse)
```

Comment:
I learned the other day about `ShouldHaveLength`

Reply:
you can also write your own ```ShouldFunctions```